apps/ledger-live-desktop/src/renderer/screens/stake/index.tsx

```diff
@@ -12,58 +12,63 @@ import { openModal } from "~/renderer/actions/modals";
 
 type Props = {
   currencies?: string[];
+  shouldRedirect?: boolean;
+  alwaysShowNoFunds?: boolean;
 };
 
-const useStakeFlow = ({ currencies }: Props = {}) => {
+const useStakeFlow = () => {
   const history = useHistory();
   const { params: { list } = { list: undefined } } = useFeature("stakePrograms") || {};
   const dispatch = useDispatch();
 
-  const cryptoCurrencies = useMemo(() => {
-    return filterCurrencies(listCurrencies(true), {
-      currencies: currencies || list,
-    });
-  }, [currencies, list]);
+  return useCallback(
+    ({ currencies, shouldRedirect = true, alwaysShowNoFunds = false }: Props) => {
+      const cryptoCurrencies = filterCurrencies(listCurrencies(true), {
+        currencies: currencies || list,
+      });
 
-  return useCallback(() => {
-    page("Stake", "Drawer - Choose Asset", {
-      ...stakeDefaultTrack,
-      page: history.location.pathname,
-      type: "drawer",
-    });
-    setDrawer(
-      SelectAccountAndCurrencyDrawer,
-      {
-        currencies: cryptoCurrencies,
-        onAccountSelected: (account: Account, parentAccount: Account | null = null) => {
-          track("button_clicked", {
-            ...stakeDefaultTrack,
-            button: "asset",
-            page: history.location.pathname,
-            currency: account?.currency?.family,
-            account,
-            parentAccount,
-            drawer: "Select Account And Currency Drawer",
-          });
-          setDrawer();
-          dispatch(openModal("MODAL_START_STAKE", { account, parentAccount }));
-          history.push({
-            pathname: `/account/${account.id}`,
-          });
+      page("Stake", "Drawer - Choose Asset", {
+        ...stakeDefaultTrack,
+        page: history.location.pathname,
+        type: "drawer",
+      });
+      setDrawer(
+        SelectAccountAndCurrencyDrawer,
+        {
+          currencies: cryptoCurrencies,
+          onAccountSelected: (account: Account, parentAccount: Account | null = null) => {
+            track("button_clicked", {
+              ...stakeDefaultTrack,
+              button: "asset",
+              page: history.location.pathname,
+              currency: account?.currency?.family,
+              account,
+              parentAccount,
+              drawer: "Select Account And Currency Drawer",
+            });
+            setDrawer();
+            dispatch(openModal("MODAL_START_STAKE", { account, parentAccount, alwaysShowNoFunds }));
```

Comment:
We could dispatch here directly to MODAL_NO_FUNDS_STAKE, avoiding move that logic too deep (MODAL_START_STAKE)
```suggestion
            dispatch(openModal("MODAL_NO_FUNDS_STAKE", { account, parentAccount }));
```
On the other hand, what do you think if instead use "alwaysShowNoFunds" we use "mode" (Thinking if tomorrow we need to create more modes....for instance "claimRewards"
```
mode: "stake"
mode: "noFunds"
```